statsmodels/discrete/tests/test_constrained.py

```diff
@@ -65,61 +65,54 @@ def test_basic(self):
 
 
     def test_basic_method(self):
```

Comment:
don't change this,
`if hasattr(self, 'res1m')` should stay in both test functions

Reply:
OK.  Are there more forthcoming cases where the attribute doesn't exist yet?

Reply:
most likely or maybe, support for constrained fitting is still missing in some models, e.g. my discrete PR to add offset and fit_constrained to Binary models.
I don't remember the details of the unit tests without going through them.

This would change the usage for subclasses, so it's not a innocent change and should stay as the original author (I) intended.